chat-plugins/calculator.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+function isNumeric(str) {
+	return !isNaN(parseFloat(str)) && isFinite(str);
+}
+
+function clean(arr) {
+	let i;
+	for (i = 0; i < arr.length; i++) {
+		if (arr[i] === "") {
+			arr.splice(i, 1);
+		}
+	}
+	return arr;
+}
+
+function parseMathematicalExpression(infix) {
+	// Shunnting-yard Algorithm -- https://en.wikipedia.org/wiki/Shunting-yard_algorithm
+	let outputQueue = "";
+	let operatorStack = [];
+	let operators = {
+		"^": {
+			precedence: 4,
+			associativity: "Right",
+		},
+		"/": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"*": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"+": {
+			precedence: 2,
+			associativity: "Left",
+		},
+		"-": {
+			precedence: 2,
+			associativity: "Left",
+		},
+	};
+	infix = infix.replace(/\s+/g, "");
+	infix = infix.split(/([+\-*/^()])/);
+	infix = clean(infix);
```

Comment:
You don't actually need a `clean` function at all.

```suggestion
	infix = infix.filter(token => token);
```

Reply:
Done.